server/bleep/src/webserver/middleware.rs

```diff
@@ -16,34 +16,73 @@ use tracing::error;
 #[derive(Serialize, Clone)]
 pub enum User {
     Unknown,
-    Authenticated {
+    Desktop {
+        api_token: String,
         login: String,
+        #[serde(skip)]
+        crab: Arc<dyn Fn() -> anyhow::Result<octocrab::Octocrab> + Send + Sync>,
+    },
+    Cloud {
+        org_name: String,
         api_token: String,
+        login: String,
         #[serde(skip)]
         crab: Arc<dyn Fn() -> anyhow::Result<octocrab::Octocrab> + Send + Sync>,
     },
 }
 
 impl User {
-    pub(crate) fn login(&self) -> Option<&str> {
-        let User::Authenticated { login, .. } = self else {
+    pub fn login(&self) -> Option<&str> {
```

Comment:
Can we rename this `id` since all it does is retrieve the user id.

Reply:
Github calls it `login`, and it's not an opaque symbol, so i'd steer away from `id`. It's also not any internal user id, but the user's visible login name as per Github.

Reply:
What about `username`? I've also found `login` confusing in the past before.